### DISCUSSION TITLE
O que se caracteriza um Sênior?

### DISCUSSION THREAD

**Discussion by willianjusten** [frontendbr/forum#252] [Discussão]
Bom galera, quero levantar essa discussão, porque acho que vai gerar bastante coisa interessante e muitos pontos de vista diferentes.

Hoje pela manhã li um excelente artigo sobre, que se chama [The conjoined triangles of senior level development](http://frontside.io/blog/2016/07/07/the-conjoined-triangles-of-senior-level-development.html) que aborda bastante coisa sobre isso.

Em breve vou postar minhas considerações, mas queria que vocês também discutissem sobre.

@frontendbr/admins respondam hein =p

**Comment by ghost** [comment]
Seria um pleno bastante experiente que compartilha seus conhecimentos e se comporta como júnior diante de outros desenvolvedores e novas ideias/tecnologias?

**Comment by viniciusdocanto** [comment]
Falei sobre isso aqui #92 também

**Comment by willianjusten** [comment]
Opa @viniciusdocanto não havia visto essa sua issue! Erro meu xD
Mas aqui queria pontuar mais só sobre a visão do Sênior mesmo, aconselho lerem primeiro o artigo, que é super interessante =D

**Comment by woliveiras** [comment]
Achei essa definição muito boa:

Nesse nível você possui muita experiência com as linguagens de programação que utiliza, com as ferramentas de trabalho, ambiente de desenvolvimento, Sistema Operacional, etc.

Agora você é capaz de criar algo do zero e escalar, sozinho!

Você tem uma boa noção de prazos e consegue especificar isso muito bem. Você também é capaz de coordenar um projeto, organizar um time, sabendo delegar tarefas, avaliar as tarefas concluídas por outros membros e responder precisamente ao cliente, gerente ou algo parecido.

Você, agora, é capaz de formar outros profissionais. Consegue direcionar eles e ajudá-los a cumprir suas metas.

Tem conhecimento do Mercado de Trabalho e consegue contratar bons profissionais ou indicá-los.

Você consegue tomar decisões cada vez mais importantes em um projeto.

Ser Senior não está ligado somente ao tempo em que você trabalha, mas ao seu conhecimento adquirido nesse tempo e as suas atitudes profissionais.

https://github.com/woliveiras/front-end-career/blob/master/translations/pt-br/sources/developer.md#sênior

**Comment by eduardojmatos** [comment]
@willianjusten @woliveiras  bacana essa discussão!

Minha opinião sobre isso é compartilhada com o que o @woliveiras disse.

Existe uma certa controvérsia sobre a definição de sênior, principalmente porque isso tende a variar muito entre as empresas.

Eu gosto de pensar que, um sênior precisa ser avaliado dentro de um time. Quando um desenvolvedor começa a ter as responsabilidades e conhecimentos que o @woliveiras citou, ou quando dentro da comunidade ele tem as mesmas habilidades, acredito que sim, ele pode ser chamado de sênior.

Isso é controverso também, porque de repente, dentro de um time de pessoas, todos podem manjar muito. Mas no fim das contas, se a responsa de cada um é igual, não vejo nenhum problema em equalizar o salário dessas pessoas.

Muitos associam "seniôridade" ao tempo de experiência. Eu acho que isso pode ser um pouco ruim na hora de avaliar o conhecimento e habilidades. Muitos tem anos de desenvolvimento, mas pouca habilidade ou de repente não se atualizou com o tempo. Associar diretamente ao tempo de experiência pode não ser 100% confiável.

Enfim, é uma discussão quase filosófica hehehe

**Comment by fernandofleury** [comment]
Acho que existem dois tipos de sênior. O sênior da empresa, e o sênior de mercado. Posso dissertar um pouco mais sobre isso se quiserem :P

**Comment by kelciocajueiro** [comment]
Aí vai minha opinião:

Se **eu** fosse contratar alguém para uma vaga de Senior, descartaria logo o critério "tempo de experiência". Conheço algumas pessoas que possuem uns bons anos no mercado mas quando eu converso sobre um assunto mais avançado (que estou estudando, por exemplo), a pessoa não consegue conversar no mesmo nível.

Muitos dizem que a "senioridade" é atingida quando o profissional vai la e faz, sem depender de alguma outra pessoa. Eu concordo em parte. Para mim o mais importante em um profissional é que ele consiga resolver o problema da melhor forma possível, ou que consiga encontrar várias formas de resolver o mesmo problema, sempre pensando em padrões, qualidade de código, etc.

**Comment by eduardojmatos** [comment]
@fernandofleury sim mano, exatamente isso. Eu gosto mais da abordagem de sênior de empresa. Mas as empresas contratam com base do "sênior de mercado", como você disse... Fale mais pra gente =D

**Comment by barbier** [comment]
Eu gosto desse artigo
https://medium.com/programador-sincero/o-que-%C3%A9-um-programdor-s%C3%AAnior-de-verdade-9b95853693d9#.5tine4j8w

**Comment by tcelestino** [comment]
@barbier, esse artigo é muito bom mesmo.

Acredito que o sênior tem não só saber das tecnologias, mas também lidar com as pessoas, até porque querendo ou não, a pessoa vai ser sempre ser a primeira a ser requisitada, seja para participar de um planejamento ou até mesmo para dúvidas mais *"infantis" do mundo.

_*infantis - perguntas básicas, mas que qualquer um pode fazer, já que ninguém sabe tudo!_ 👍

**Comment by felipeuntill** [comment]
Não acredito em níveis profissionais, comecei a programar enquanto criança e quando entrei no mercado  de trabalho como Junior, já desenvolvia melhor que o suposto desenvolvedor Sênior da empresa onde trabalhava em questão de 6 meses era nítido que eu tinha mais capacidade e conhecimento e melhores entregas. Exatamente um ano depois, decidi pedir aumento uma vez que o mercado estava pagando salários até 4 vezes superiores aos que eu estava recebendo, foi com grande espanto que recebi a seguinte informação:
-'_Você é muito novo para receber este valor, isso é oque estamos pagando para o "Fulano Sênior" que tem o dobro da sua idade_.
Questionei sobre o fato de que minha qualidade técnica era superior e minhas entregas eram significativamente melhores e mesmo assim indagaram que o salário não era compatível com minha idade. Resumindo nesta empresa em específico ser Sênior significava ser mais velho independendo da qualidade técnica. 
OBS: como trabalhávamos em uma agência não existia conhecimento de negócio uma vez que não eramos especializados em nada.

Anos depois comecei a atuar com comércio eletrônico e em período de meses por apresentar um fácil entendimento das regras de negócio dos clientes e tecnologias utilizadas pela empresa fui promovido a Sênior e cheguei a ficar responsável por uma equipe de 25 recursos, alguns com o dobro da minha idade tendo somente 21 anos de idade. Em suma, esta empresa ignorava o fator idade e prezava pela impotência do profissional dentro do time, cenários similares persistiram nos meus 3 próximos empregos.

2 anos atrás fui convidado e trabalhar no Bradesco atuando nas integrações com as plataformas de comércio eletrônico, uma vez eu já possuía significativa experiência neste nicho, por ser a maior entidade da equipe no que competia a comércio eletrônico a diretoria titulou especialista titulo que normalmente era concebido a pessoas que já estavam a anos dentro da empresa e possuíam significativamente mais idade nesta época eu estava próximo dos 25 anos de idade e acreditem este evento causou uma série de problemas, alguns profissionais pediram transferência da área, outros se recusavam a ficar subordinados a mim em determinados projetos. Nesta oportunidade a maior resistência era dos membros do time que não se sujeitavam a ficar abaixo de uma pessoa mais nova.

Hoje eu estou novamente atuando com comércio eletrônico, não fui contratado sobre qualquer denominação de Junior, Pleno ou Sênior mas estou abaixo de 2 níveis hierárquicos, minha liderança técnica e meu gestor, meu salário é significativamente superior ao de ambos portanto nesta empresa a remuneração não está relacionada ao nível organizacional.

Destacando as experiências citadas acima eu aprendi a ignorar rótulos, eu acredito na precificação do meu conhecimento, se eu souber que existe um recurso ganhando mais do que e entregando menos eu vou pedir um aumento(no tempo devido) ou vou mudar de emprego, posso ser descrito como estagiário desde que me paguem oque eu pedir. No final, **everithing is about money**

**Comment by akfzambrana** [comment]
O artigo que o @barbier linkou é interessante, tem algumas partes ligeiramente arbitrarias, especialmente quando fala sobre educação, pois existem excelentes profissionais, seniores na vida e na profissão, que foram pra faculdades mais "humirdes" que as que ele cita como "boas", no entanto para toda regra existe alguma exceção e eu posso estar sendo arbitraria na minha opinião também.
Em uma conversa durante um FEMUG-SP surgiu esta mesma discussão e algumas coisas que foram ditas lá eu levei para minha vida. Acredito (depois dessa discussão) que o que faz a pessoa ser sênior é uma mistura de **caráter + conhecimento + experiência**.
Primeiro você adquire o conhecimento, estudando sozinho, na faculdade, no curso técnico, na internet, etc. daí com o tempo você começa a ganhar experiencia, mas não falo de tempo de experiencia e sim de "coisas experimentadas" (bugs, problemas, desafios, etc) e ate ai você já consegue virar pleno e se manjar muito de tecnologia até consegue ser o(a) ninja das galaxias seja na tua empresa, na comunidade ou no mercado. 
Infelizmente ser o(a) ninja das galaxias e além, não significa ser sênior, pelo menos não no meu conceito, alem de você ter tudo isso já citado acima você precisa ter **caráter** e nessa parte o artigo do "programador sincero" esta um pouco mais correto. Um(a) desenvolvedor(a) sênior, precisa ter autocontrole, ter uma certa didática para compartilhar seu conhecimento, saber ouvir mais do que falar, ser humilde ao ponto de aceitar a opinião dos outros (acho que muitos de nós pecamos neste quesito), ter responsabilidade encima do código (seu e dos outros), saber mostrar sua opinião sem ser arrogante e autoritário(a), saber quando dar carteirada (as vezes é preciso, infelizmente), saber lidar com pessoas de todos os níveis sociais e de conhecimento e se manter sempre atualizado com respeito à tecnologia. Alto super importante neste conceito é que um(a) desenvolvedor(a) senior não precisa saber tudo sobre tecnologia (pra isto existe o Google e as documentações). Praticamente um(a) senior é quase um dalai lama da tecnologia, misturando tudo em algo maior do que apenas um "label".
Minha humilde contribuição à discussão :kissing_heart:

**Comment by marcosalberto** [comment]
@felipeuntill concordo com tudo que falou, o problema é que para as pessoas que estão no início da carreira esses títulos ajudam a se posicionar no mercado. E pelo lado da empresa, ajuda a filtrar quem esta apto a preencher a vaga. Acho que esse é o modo prático de agir como profissional, inclusive onde estou não tenho isso definido, mas os títulos ainda são úteis.

**Comment by felipeuntill** [comment]
@marcosalberto Discordo, cansei de perder meu tempo entrevistando Sênio bom de bico que não produz uma linha de código sequer, claro que também cansei de ver Junior que resolve tudo mas não sabe abrir a boca pra falar! 

Cada contratação precisa de um contexto e para cada contexto existe um perfil de skillset e mindset adequados. 

Os testes e entrevistas exatamente servem pra balizar a expectativa do contratador o currículo mente a execução nunca deixa passar nada.

**Comment by willycamargo** [comment]
Não consigo entender a necessidade de nivelar os profissionais entre "Junior, Pleno, Senior, Ninja, Jedi etc..."

O real é que você precisa ser bom no que faz. Como discutido em outras issues (ex #32 e  #92) existem diversos tipos de profissionais e níveis de conhecimento.

Você tem que atender as necessidades da empresa, sabendo lidar com a linguagem/framework que trabalha de forma independente. Ser criativo a ponto de resolver problemas de forma inovadora e simples, além de saber se comunicar com os colegas de trabalho e com a comunidade a fim de estar absorvendo as novidades relacionadas as suas atividades cotidianas e ao stack do projeto.

Eu talvez possa ser considerado Senior no meu cargo atual para a empresa que trabalho, porém, se me tirar daqui, onde trabalho especificamente com MithrilJS e ReactJS e me colocar em outro  lugar que trabalhe com AngularJS lá eu poderei ser considerado Junior. Mas minha experiência com a linguagem (javascript) e padrões de desenvolvimento me permitirá uma curva de aprendizagem que pode me levar de "Junior" para "Senior" em alguns meses.

E então, como categorizar "Junior, Pleno e Senior", se na verdade antes de mandar a pessoa sentar e trabalhar você não sabe como ela vai se sair com os problemas cotidianos?

Acredito que o ideal na verdade é o desenvolver escolher a empresa na qual quer trabalhar, e não se basear apenas em salário para isso.

---

Imaginemos dois desenvolvedores:

João: desenvolvedores front que prefere trabalhos de agência. Do tipo que em 15 dias está pronto, e tendo qualificação pra isso esse front pode ser considerado "Senior" no ambiente em que trabalha, e assim garantir um salário significante.

Mauro: profissional que prefere trabalhar em um único sistema (qualquer tipo de SaaS), e estar desenvolvendo features e melhorias progressivas nesse mesmo sistema. Se esse desenvolvedor sabe trabalhar com escalabilidade, escrever um código de qualidade e manter um bom padrão sem se afundar em problemas com o passar do tempo, esse desenvolver pode ser considerado Senior no ambiente em que trabalha.

Agora coloque o Mauro para trabalhar no lugar de João, e João no lugar do Mauro.

Se João escrever código rápido e que não apresente tanta qualidade e conseguir entregar 3x mais rápido, em algumas semanas estará afundado em problemas que ele mesmo criou e o sistema deverá ser **refeito** do zero.

Se Mauro escrever código de qualidade, com bom design patterns e pronto para ser escalável, seria fantástico, mas ele conseguiria entregar um projeto a cada 45 dias.

Bom, Mauro é Junior em um agência e merece receber salário de Junior lá. José é Junior trabalhando com um SaaS, e merece receber salário de Junior fazendo isso.

**Comment by renatoagds** [comment]
Acho algo bem complicado essa separação entre "Junior, Pleno, Sênior, etc", pois vamos esbarrar em muitas opiniões controversas e pessoais, mas deixando a minha, acho que determinar alguém como Sênior, como alguns disseram aqui, vai muito além de conhecimento técnico, acredito que o cara Sênior tem que ser alguém que pode ser ponto de referência, em todos os âmbitos (técnico, caráter, business, etc), e esse conjunto, na minha visão, faz dele um Sênior.

Ao longo da minha experiência já vi caras que eram caracterizados como Sênior apenas pelo conhecimento técnico, mas no dia-a-dia não conseguiam agregar valor, pois no conjunto deixavam a falhar, assim como caras que não tinham um conhecimento alto, mas agregavam muito valor pelo conjunto de skills que ele tinha.

Com isso acho que cabe a empresa definir esse conjunto de skills que determinam alguém Sênior e ir no mercado procurar alguém que se encaixe nisso que ela procura, pois volto a dizer, não acho que Sênior é algo que se pode bater o martelo para X ou Y coisas.

**Comment by barbier** [comment]
@akfzambrana eu não concordo em algumas coisas do texto, sobre educação, um dos melhores programadores que eu conheço é biólogo(!), mas em geral eu gosto do material postado por ele...

**Comment by felipefialho** [comment]
@willycamargo Sua visão faz todo sentido.

**Comment by willianjusten** [comment]
Eu gosto bastante do texto que passei, pois ele separa alguns pontos que acho válidos:
- Provide clear technical and project guidance for their teammates and our clients.
- Mentor, teach, and contribute internally and in larger programming communities.
- Ship software that is a joy to its users and to the developers who extend and maintain it.

Que também é representada nessa imagem:

![image](https://cloud.githubusercontent.com/assets/3991845/17121435/19a78ca0-52aa-11e6-8261-060f5cff9b65.png)

O sênior é aquele que consegue juntar **liderança técnica nos projetos**, **sabe passar conhecimento e contribui com comunidade** e **conhecimento técnico**.

Junto a isso, eu acho válido colocar o conhecimento do Produto em que está ou das regras de negócio do que trabalha.

Muito do que foi dito acima eu também concordo, com algumas ressalvas. Eu acho que é válido sim ter níveis hierárquicos, até mesmo para a pessoa ver evolução, se ela num souber se é Jr, Pleno, Sênior, Especialista, seja lá o que for, ela não vai ter muitas ambições e nem para onde crescer.

Mas eu acho válido que a análise de nível seja feita no processo e não que tenha a vaga já determinada "vaga de senior blablabla", porque as pessoas não podem conseguir diferenciar se são ou não do nível pedido.

**Comment by akfzambrana** [comment]
Sim @willianjusten, concordo com você e acredito que essa imagem resume perfeitamente, inclusive o que eu penso também heheh, espero um dia conseguir chegar a sr :smile:
